### PR TITLE
Add ability to output compiler performance information

### PR DESCRIPTION
**Pull request by kitsonk**
This PR add the flag `--performance` to sub commands like `run`, `test`, `bundle`, and `cache`.  When this done, output looks something like this:

```
$ deno run --reload --performance cli/tests/002_hello.ts
Compile file:///Users/kkelly/github/deno/cli/tests/002_hello.ts
  Files: 43
  Nodes: 28595
  Identifiers: 11251
  Symbols: 8029
  Types: 3570
  Instantiations: 1064
  Parse time: 54
  Bind time: 905
  Check time: 1847
  Emit time: 20
  Total TS time: 2826
  Compile time: 2866
Hello World
```

In addition, when `--log-level` is set to `debug` the stats will be output in the file.

The PR includes a few other minor improvements that were noted, like getting rid of a couple very noisy debug lines that weren't informative and aligning the output of `deno bundle` to be consistent with other subcommands.

### CODE REVIEW

**Comment by bartlomieju** [comment]
@kitsonk I'd suggest showing this output always when `-Ldebug` flag is present. We can add --performance flag later - that way we won't have to wait for v1.2.0 to land this PR

**Comment by kitsonk** [comment]
Ok, I will strip it back to remove `--performance` out of this PR.

File: cli/js/compiler.ts [hunk 1]
```diff
@@ -843,6 +847,46 @@ const ignoredDiagnostics = [
...
+  stats.push({ key: "Types", value: program.getTypeCount() });
+  stats.push({ key: "Instantiations", value: program.getInstantiationCount() });
+  const programTime = ts.performance.getDuration("Program");
+  const bindTime = ts.performance.getDuration("Bind");
+  const checkTime = ts.performance.getDuration("Check");
+  const emitTime = ts.performance.getDuration("Emit");
+  stats.push({ key: "Parse time", value: programTime });
+  stats.push({ key: "Bind time", value: bindTime });
+  stats.push({ key: "Check time", value: checkTime });
+  stats.push({ key: "Emit time", value: emitTime });
```
Comment (ry) [inline cli/js/compiler.ts:876]: Use the same keys as used in TS API?
So `s/Emit time/Emit/` and `s/Check time/Check/` and so on.

File: cli/js/compiler.ts [hunk 2]
```diff
@@ -843,6 +847,46 @@ const ignoredDiagnostics = [
...
+  stats.push({
+    key: "Total TS time",
+    value: programTime + bindTime + checkTime + emitTime,
+  });
```
Comment (ry) [inline cli/js/compiler.ts:880]: Maybe leave this one out since it doesn't come directly from TS

Comment (kitsonk) [reply cli/js/compiler.ts:880]: This mirrors what the `tsc` puts out as "Total time" (same calculation), if we wanted to compare apples to apples.

**Comment by David-Else** [comment]
I suggest adding `ms` for units of time, unless `Emit time: 20` is 20 easter eggs or hamsters :) 

Adding `,` for 1,000,000,000 would be nice too for human readability.

**Pull request merged**